arviz/tests/base_tests/test_data.py

```diff
@@ -1111,6 +1112,23 @@ def test_empty_inference_data_object(self):
         assert not os.path.exists(filepath)
 
 
+class TestJSON:
+    def test_json_converters(self, models):
+        idata = models.model_1
+
+        filepath = os.path.realpath("test.json")
+        idata.to_json(filepath)
+
+        idata_copy = from_json(filepath)
+        for group in idata._groups_all:  # pylint: disable=protected-access
```

Comment:
should we add an `__iter__` method to inferencedata or use a dict_like behaviour? looping over groups seems like something that should be possible and part of the public api. cc @ahartikainen @aloctavodia @canyon289 @AlexAndorra

Reply:
Yes, that sounds reasonable.

Reply:
Cool, ai think dict like is what will be more intuitive so users can choose items, keys or values, maybe renaming keys to groups?

It could be done in another PR, just thought of this here

Reply:
Yes, I'll implement that!